test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -152,18 +192,340 @@ def test_simple_shock_upload(self):
         ### Test for upload from SHOCK - upload the file to shock first
         print('attempting upload through shock')
         gbk_path = "data/e_coli/GCF_000005845.2_ASM584v2_genomic.gbff"
-        data_file_cli = DataFileUtil(os.environ['SDK_CALLBACK_URL'])
         shutil.copy(gbk_path, self.cfg['scratch'])
-        shock_id = data_file_cli.file_to_shock({
+        shock_id = self.dfuClient.file_to_shock({
             'file_path': os.path.join(self.cfg['scratch'], gbk_path.split("/")[-1])
         })['shock_id']
         print("Running test")
         ws_obj_name2 = 'MyGenome.2'
-        result = self.getImpl().genbank_to_genome(self.getContext(), {
+        result = self.serviceImpl.genbank_to_genome(self.ctx, {
                 'file': {'shock_id': shock_id},
-                'workspace_name': self.getWsName(),
+                'workspace_name': self.wsName,
                 'genome_name': ws_obj_name2,
             })[0]
         self.assertIsNotNone(result['genome_ref'])
         self.assertTrue(int(
             result['genome_info'][10]['Number of Protein Encoding Genes']) > 0)
+
+    def _run_test_fail(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbank_to_genome(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _run_test_fail_mass(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbanks_to_genomes(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _load_expected_data(self, json_path):
+        with open(json_path, "r") as read_file:
+            data = json.load(read_file)
+        return data
+
+    def _retrieve_provenance(self, provenance):
+        # make a copy to avoid modifying the original provenance
+        provs = [prov.copy() for prov in provenance]
+        for prov in provs:
+            for key in ["time", "epoch"]:
+                prov.pop(key)
+            for subaction in prov['subactions']:
+                subaction.pop("commit")
+        return provs
+
+    def _retrieve_genome_metadata(self, metadata):
+        # make a copy to avoid modifying the original metadata
+        metadata = dict(metadata)
+        metadata.pop("Assembly Object")
+        return metadata
+
+    def _retrieve_genome_data(self, data):
+        # make a copy to avoid modifying the original genome data
+        data = dict(data)
+        for key in ["assembly_ref", "genbank_handle_ref"]:
+            data.pop(key)
```

Comment:
Can you check these look like handle IDs

Reply:
How do I check is handle IDs? I delete genbank_handle_ref because is changing for different run.

Reply:
Now I think about it, what you should do is:

* look up the handle in the handle service
* pull the blobstore node using the shock id in the handle
* check the file name and md5 are correct in the blobstore node

Reply:
Please correct me if I am wrong, 
HandleService.hids_to_handles to get `shock_id` and then dfu.shock_to_file to get `blobstore node`?

Reply:
No need to pull the file, just made a direct call to the blobstore to get the node and check the filename and md5 is what you expect

Reply:
any shock_id in CI I can use to experiment? Not sure how to find it in kbase UI

Reply:
Not totally sure what you're asking, but there's shock / blobstore urls for the images in collections

Reply:
I assume `making a direct call to the blobstore` means send a GET request to https://ci.kbase.us/services/blobstore/node/<shock_id>. And I want to find a shock_id in CI to experiment it.

Reply:
Correct, and collections images will have url like that

Reply:
ah, do I need to include `token` when making a direct call to get the blobstore node? I got `User Unauthorized` error.

Reply:
Not for the urls in collections, they should be public. For the ones in the tests that were created by the test user, yeah you do: https://github.com/kbase/blobstore?tab=readme-ov-file#api

Reply:
but it says optional?

```
AUTHORIZATION OPTIONAL
GET /node/<id>

RETURNS: a Node.
```

Reply:
so the default is private unless we make it public?

Reply:
Yep, that's the safest route and what users would generally expect

Reply:
The blobstore md5 is the md5 of the file that was stored in the blobstore, so either the fasta or genbank file

Reply:
The blobstore calculates the md5 from the file contents

Reply:
which line? I need to see how it converted in GO, and do the same thing in Python and then compare corresponding md5

Reply:
Why not just do an md5sum of the file?

Reply:
still != '22e84120aefac5f46a4ffa5de3ec0478'. I did md5 sum of `Cyanidioschyzon_merolae_one_locus.gbff`
<img width="783" alt="Screenshot 2024-04-25 at 4 01 58 PM" src="https://github.com/kbaseapps/GenomeFileUtil/assets/22202096/5b910cfe-e6d3-4b33-93c5-45f235f08142">

I need to see in GO how it is converted.

Reply:
what does md5sum say the md5 is?

Reply:
maybe we're not saying the same thing here https://en.wikipedia.org/wiki/Md5sum

Reply:
<img width="932" alt="Screenshot 2024-04-25 at 4 26 57 PM" src="https://github.com/kbaseapps/GenomeFileUtil/assets/22202096/64e5220e-c2eb-44a7-afb0-a1f41059e189">

Reply:
What's the blobstore say?

Reply:
```
blob_info is: {'data': {'attributes': None, 'created_on': '2024-04-25T20:09:22.000Z', 'file': {'checksum': {'md5': '22e84120aefac5f46a4ffa5de3ec0478'}, 'name': 'Cyanidioschyzon_merolae_one_locus.gbff.gz', 'size': 28818}, 'format': '', 'id': '856945bd-fec6-4386-a9f1-83f445407a11', 'last_modified': '2024-04-25T20:09:22.000Z'}, 'error': None, 'status': 200}
```

Reply:
The shock file is gzipped

Reply:
genbank files got preprocessed before uploading to blobstore. I need to download and run md5sum on them instead.

Reply:
This is confusing. I run the same script again. 'md5' is changing so it means this `md5` cannot be just file content.

```
blob_info is: {'data': {'attributes': None, 'created_on': '2024-04-26T23:29:17.000Z', 'file': {'checksum': {'md5': 'f42fc6df9f05a74edc6395345780ad87'}, 'name': 'Cyanidioschyzon_merolae_one_locus.gbff.gz', 'size': 28818}, 'format': '', 'id': '1a71eaca-cc91-4daf-9735-39e2e34427b2', 'last_modified': '2024-04-26T23:29:17.000Z'}, 'error': None, 'status': 200}
```

Reply:
It's because it's gzipped. Gzipped files contain a time stamp by default and so the md5 won't be repeatable. You'll have to download the file and calculate the md5

Reply:
so it should be:

- look up the handle in the handle service
- download the blobstore node using the shock id in the handle
- calculate the md5 and compare with that of preprocessed genbank file

Reply:
I assume the processed file is what gets uploaded right? So that comparison won't work

Reply:
But otherwise yes

Reply:
Good catch, updated above.

Reply:
I downloaded the file content using [this](https://github.com/kbase/blobstore?tab=readme-ov-file#download-a-file-from-a-node), and then calculate md5sum of the response.text. Got `a4aeeabc5842bbcd91ae8d2805497a7a`, which is != md5sum of processed file

Reply:
Did you unzip the file?

Reply:
unzip the downloaded file from blobstore? I made a direct call using [this](https://github.com/kbase/blobstore?tab=readme-ov-file#download-a-file-from-a-node) and I got content string directly. Not sure which step I should unzip the file.

Reply:
The file in the blobstore is zipped, so the md5 of the file you downloaded from the blobstore is not going to match the non-zipped file you're comparing it to

Reply:
Am I using the right API? Like I said, [this](https://github.com/kbase/blobstore?tab=readme-ov-file#download-a-file-from-a-node) returned the content string directly and I have no file to unzip.

Reply:
The content string (really bytes) is the file.

Reply:
👍

Reply:
It looks like the filename in the blobstore isn't checked here either

Reply:
👍

Reply:
: